streaming/src/main/scala/org/apache/spark/streaming/scheduler/BatchInfo.scala

```diff
@@ -33,7 +33,7 @@ import org.apache.spark.streaming.Time
 @DeveloperApi
 case class BatchInfo(
     batchTime: Time,
-    streamIdToNumRecords: Map[Int, Long],
```

Comment:
Okay I chatted with @pwendell , and the decision in as follows.

1. Name = StreamInputInfo, 

2. It will have the field info: Map[String, Object]. Rather than a string, its most future-proof is to use a Map[String, Object], as we may want to not just strings but actual objects. For example, for files, one of the keys in the map be "files" and the value is a list of files.

3. Deprecate streamIdToNumRecords and introduce streamIdToInputInfo: Map[Int, StreamInputInfo]

4. object StreamInputInfo can have well known key names like for number of records, etc. They are used to store the corresponding data in the map.